homeassistant/components/gpsd.py

```diff
@@ -0,0 +1,130 @@
+"""
+Support for GPSD.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/gpsd/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.util as util
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util import dt as dt_util
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (ATTR_LATITUDE, ATTR_LONGITUDE, STATE_UNKNOWN)
+
+REQUIREMENTS = ['gps3==0.33.2']
+
+DOMAIN = "gpsd"
+ENTITY_ID = "{}.gps".format(DOMAIN)
+
+CONF_HOST = 'host'
+CONF_PORT = 'port'
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 2947
+
+ATTR_GPS_TIME = 'gps_time'
+ATTR_ELEVATION = 'elevation'
+ATTR_SPEED = 'speed'
+ATTR_CLIMB = 'climb'
+ATTR_MODE = 'mode'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setup the GPSD component."""
+    conf = config[DOMAIN]
+    host = util.convert(conf.get(CONF_HOST), str, DEFAULT_HOST)
+    port = util.convert(conf.get(CONF_PORT), int, DEFAULT_PORT)
+
+    # Will hopefully be possible with the next gps3 update
+    # https://github.com/wadda/gps3/issues/11
+    # from gps3 import gps3
+    # try:
+    #     gpsd_socket = gps3.GPSDSocket()
+    #     gpsd_socket.connect(host=host, port=port)
+    # except GPSError:
+    #     _LOGGER.warning('Not able to connect to GPSD')
+    #     return False
+    import socket
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        sock.connect((host, port))
+        sock.shutdown(2)
+        _LOGGER.debug('Connection to GPSD possible')
+    except socket.error:
+        _LOGGER.warning('Not able to connect to GPSD')
+        return False
+
+    gpsd = Gpsd(hass, host, port)
+    gpsd.point_in_time_listener(dt_util.utcnow())
```

Comment:
For me the whole "GPS thing" was always more similar to how we handle the sun than what sensors do. Per definition we are just observing various states, so it should be a sensor. I will change it.